docs/blog/2018-12-17-turning-the-static-dynamic/index.md

```diff
@@ -0,0 +1,405 @@
+
+## 5 Steps to add Netlify Identity and Authenticated Pages to Gatsby
+
+1. **Enable Netlify Identity**: Netlify Identity doesn't come enabled by default. You'll have to head to your site admin (eg `https://app.netlify.com/sites/YOUR_AWESOME_SITE/identity`) to turn it on. [Read the docs](https://www.netlify.com/docs/identity/) for further info on what you can do, for example add Facebook or Google social sign-on!
+2. **Install dependencies**: `npm install netlify-identity-widget gatsby-plugin-create-client-paths`
+3. **Configure Gatsby**: for dynamic-ness!
+
+```js
+// gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-plugin-create-client-paths`,
+      options: { prefixes: [`/app/*`] },
+    },
+    // ...
+  ],
+  // ... (including what you also wrote earlier)
+}
+```
+
+4. **Write an authentication service**: `netlify-identity-widget` is a framework-agnostic overlay that ships with a nice signup/login UI. This gets you up and running the fastest, however if you need a smaller solution you may want to use the underlying [gotrue-js](https://github.com/netlify/gotrue-js), or [react-netlify-identity](https://github.com/sw-yx/react-netlify-identity) for a React Hooks solution.
+
+Here's a usable example that stores your user in local storage:
+
+```js
+// service/auth.js
+import netlifyIdentity from "netlify-identity-widget"
+
+export const isBrowser = () => typeof window !== "undefined"
+export const initAuth = () => {
+  if (isBrowser()) {
+    window.netlifyIdentity = netlifyIdentity
+    // You must run this once before trying to interact with the widget
+    netlifyIdentity.init()
+  }
+}
+export const getUser = () =>
+  isBrowser() && window.localStorage.getItem("netlifyUser")
+    ? JSON.parse(window.localStorage.getItem("netlifyUser"))
+    : {}
+
+const setUser = user =>
+  window.localStorage.setItem("netlifyUser", JSON.stringify(user))
+
+export const handleLogin = callback => {
+  if (isLoggedIn()) {
+    callback(getUser())
+  } else {
+    netlifyIdentity.open()
+    netlifyIdentity.on("login", user => {
+      setUser(user)
+      callback(user)
+    })
+  }
+}
+
+export const isLoggedIn = () => {
+  if (!isBrowser()) return false
+  const user = netlifyIdentity.currentUser()
+  return !!user
+}
+
+export const logout = callback => {
+  netlifyIdentity.logout()
+  netlifyIdentity.on("logout", () => {
+    setUser({})
+    callback()
+  })
+}
+```
+
+5. **Write your app**: Now, any sub paths in `src/pages/app` will be exempt from Gatsby static generation. To keep the dividing line between app and site crystal clear, I like to have all my dynamic Gatsby code in a dedicated `app` folder. This means you can use `@reach/router` with `netlify-identity-widget` to write a standard dynamic React app. Here's some sample code to give you an idea of how to hook them up:
+
+`app.js`:
```

Comment:
this looks great. We also have a new thing where you can name the file using

```shell
```jsx:title=gatsby-config.js
```
at the top of the code snippet. We haven't implemented this as standard in blog posts, so just optional!

Reply:
Also @amberleyromo just implemented some type tab markers! I don't know how to do that yet though...see https://github.com/gatsbyjs/gatsby/pull/10515

Reply:
oh that would be nice. how would i know if that code snippet header thing is implemented or not?

Reply:
It already is for docs, for sure. The blog directory is within the docs directory, so it should work as far as I know.